Sources/SwiftParser/Specifiers.swift

```diff
@@ -797,20 +798,64 @@ extension Parser {
         unexpected.append(misspelledThrows)
         if effectSpecifiers?.throwsClause == nil {
           if matchedSubset == S.CorrectThrowsTokenKinds.self {
-            synthesizedThrows = missingToken(spec)
+            let synthesizedThrows = missingToken(spec)
+
+            var virtualParsedOffset = currentToken.start
+            synthesizedThrowsClause = self.virtualParse { this in
+              this.presence = .missing
+              defer {
+                this.presence = .present
+              }
+
+              guard let leftParen = this.consume(if: .leftParen) else {
+                return RawThrowsClauseSyntax(
+                  throwsSpecifier: synthesizedThrows,
+                  leftParen: nil,
+                  type: nil,
+                  rightParen: nil,
+                  arena: this.arena
+                )
+              }
+
+              let type = this.parseType()
+              guard case let (_, rightParen?) = this.expect(.rightParen) else {
+                return RawThrowsClauseSyntax(
+                  throwsSpecifier: synthesizedThrows,
+                  leftParen: nil,
+                  type: nil,
+                  rightParen: nil,
+                  arena: this.arena
+                )
+              }
+
+              virtualParsedOffset = this.currentToken.start
+              return RawThrowsClauseSyntax(
+                throwsSpecifier: synthesizedThrows,
+                leftParen: leftParen,
+                type: type,
+                rightParen: rightParen,
+                arena: this.arena
+              )
```

Comment:
I tried to use a lookahead here to "virtually" parse the throwing type but lookahead methods don't return any `RawSyntax` ;(
I've resorted to rewinding `currentToken` and `lexemes` to emulate virtual parsing which I think is super ugly.